scripts.js

```diff
@@ -0,0 +1,323 @@
+console.log(numbers); // [1, 2, 3]
+console.log(numbers !== squaredNumbers); // true
+
+// 4
+function getReversedString(string) {
+  let characters = string.split('');
+  characters.reverse();
+
+  return characters.join('');
+}
+
+console.log(getReversedString('Hello!')); // '!olleH'
+console.log(getReversedString('Arrays')); // 'syarrA'
+
+//5
+function isPalindrome(string) {
+  string = string.replace(/\s/g, '');
+
+  return string.toLowerCase() === getReversedString(string.toLowerCase());
+}
+
+console.log(isPalindrome('Kayak')); // true
+console.log(isPalindrome('Racecar')); // true
+console.log(isPalindrome('Was it a cat I saw')); // true
+console.log(isPalindrome('Hello!')); // false
+
+//6
+function countLetters(string) {
+  let countedLetters= {};
+  let characters = string.replace(/\s/g, '');
+  characters = characters.toLowerCase();
+  characters = characters.split('');
+
+  for (let character of characters) {
+    countedLetters[character] = (countedLetters[character] || 0) + 1;
+  }
+
+  return countedLetters;
+}
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+console.log(lettersObject);
+
+// 7
+function getYoungestPerson(peopleArray) {
+  let youngestPerson = 100;
+
+  for (let i = 0; i < peopleArray.length; i++) {
+    if (youngestPerson > peopleArray[i].age) {
+      youngestPerson = peopleArray[i].age;
+    }
+  }
+
+  return youngestPerson;
+}
+
+function getOldestPerson(peopleArray) {
+  let oldestPerson = 0;
+
+  for (let i = 0; i < peopleArray.length; i++) {
+    if (oldestPerson < peopleArray[i].age) {
+      oldestPerson = peopleArray[i].age;
+    }
+  }
+
+  return oldestPerson;
+}
+
+function getAgeDifference(peopleArray) {
+  return getOldestPerson(peopleArray) - getYoungestPerson(peopleArray);
+}
+
+const peopleArray = [
+  {
+    name: 'Adam',
+    age: 20
+  },
+  {
+    name: 'Amanda',
+    age: 5
+  },
+  {
+    name: 'John',
+    age: 75
+  },
+  {
+    name: 'Dave',
+    age: 15
+  }
+]
+
+console.log(getAgeDifference(peopleArray)); // 70
+
+// 8
+function positiveSum(numbers) {
+  let numbersSummary = 0;
+
+  for (let i = 0; i < numbers.length; i++) {
+    if (numbers[i] > 0) {
+      numbersSummary = numbersSummary + numbers[i];
+    }
+  }
+
+  return numbersSummary;
+}
+
+console.log(positiveSum([1, 5, 10]));
+console.log(positiveSum([-1, -5, -10]));
+
+// 9
+function squaredNumbersSummary(numbers) {
+  let total = 0;
+
+  for (let i = 0; i < numbers.length; i++) {
+    total = total + numbers[i] * numbers[i];
+  }
+
+  return total;
+}
+
+console.log(squaredNumbersSummary([0, 3, 4, 5]));
+
+// 10
+function abbrevName(name){
+  let initials = name[0] + '.';
+
+  for (let i = 0; i < name.length; i++) {
+    if (name[i] === ' ') {
+      initials = initials + (name[i + 1]);
+    }
+  }
+
+  return initials.toUpperCase();
+}
+
+console.log(abbrevName('Kuba klin'));
+
+// 11
+function countSheeps(sheep) {
+  let presenetSheep = 0;
```

Comment:
I think there is a typo here

Reply:
I thought that it was about the plural name of word 'sheep'. Fixed both now